daemons/controld/controld_join_client.c

```diff
@@ -325,6 +345,14 @@ do_cl_join_finalize_respond(long long action,
 
         free_xml(tmp1);
 
+        /* Update the remote node cache with information about which node
+         * is hosting the connection.
+         */
```

Comment:
It might be useful to have a comment here with the format of the XML message with all the remote info.